relay-cardinality/src/redis/state.rs

```diff
@@ -82,6 +82,11 @@ impl<'a> LimitState<'a> {
         self.scopes.entry(scope).or_default().push(entry)
     }
 
+    /// Returns `true` if this state does not contain any entries.
+    pub fn is_empty(&mut self) -> bool {
+        self.scopes.is_empty()
```

Comment:
Under what circumstances is `self.scopes` empty?

Reply:
We build the state for all limits, it is empty:

1) if there are no buckets that match a limit (e.g. due to the namespace filter), 
2) if all buckets were already handled by the cache

I'll add it to the doc string!